### PR TITLE
fix: fix and refactor `--no-lockfile-update`

### LINKED ISSUE #1669: Cannot run remove --no-install
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
[project]
dependencies = [
    "package-1",
]
```
Run `pixi remove --no-install --pypi package-1`

### Issue description

I cannot run pixi remove --no-install because i m getting:
```
  × Cannot update pypi dependencies without first installing a conda prefix that includes python.
```

I want the --no-install because the idea is to be able to run something that removes private packages and then add them as editable package instead.

```
   pixi remove --no-install --pypi private-package-1 \
    && pixi remove --no-install --pypi --feature env1 private-package 2 \
    && pixi add --pypi --pypi-keyring-provider subprocess \
    "private-package1 @ file:///app/private-package1" \
    --editable \
```

Shouldnt --no-install just clean the pyproject.toml and the pixi.lock?
Not sure where that error is coming from

### Expected behavior

I have a current workaround using sed to comment out the private packages that I do not want, and then run pixi add, but its not really clean.

### PR DESCRIPTION
**Pull request by ruben-arts**
Fixes #1669 

Refactors the cli configuration into:
- `ProjectConfig`: just the manifest path for now but resusing the cli field
- `PrefixUpdateConfig`: All settings related to the update of the prefix and lockfile
- `DependencyConfig`: All the settings to specify what type of dependency you want to interact with.

### CODE REVIEW

**Comment by ruben-arts** [comment]
Conflicts are for after the weekend when the mind is fresh

File: src/cli/remove.rs [hunk 1]
```diff
@@ -58,13 +66,13 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     // updating prefix after removing from toml
     get_up_to_date_prefix(
```
Comment (olivier-lacroix) [inline src/cli/remove.rs:67]: Shouldn't this be entirely skipped when `--no-lockfile-update` is used?

Comment (olivier-lacroix) [reply src/cli/remove.rs:67]: Otherwise, I am not sure https://github.com/prefix-dev/pixi/issues/1669 will actually be fixed?

File: pixi.toml [hunk 2]
```diff
@@ -95,3 +95,6 @@ lint = { features = [
 schema = { features = [
   "schema",
 ], no-default-feature = true, solve-group = "default" }
+
+[dependencies]
+boltons = ">=24.0.0,<25"
```
Comment (baszalmstra) [inline pixi.toml:100]: Do we need this?

File: src/cli/list.rs [hunk 3]
```diff
@@ -52,20 +52,15 @@ pub struct Args {
...
     #[arg(short, long)]
     pub environment: Option<String>,
 
     #[clap(flatten)]
-    pub lock_file_usage: super::LockFileUsageArgs,
-
-    /// Don't install the environment for pypi solving, only update the lock-file if it can solve without installing.
-    #[arg(long)]
-    pub no_install: bool,
+    pub prefix_update_config: PrefixUpdateConfig,
```
Comment (baszalmstra) [inline src/cli/list.rs:63]: This removes the `--frozen` and `--locked`.

**Review by baszalmstra — COMMENTED** [review]
I think we should remove `--no-lock-update` as a whole because I think its very confusing.

**Comment by olivier-lacroix** [comment]
> I think we should remove `--no-lock-update` as a whole because I think its very confusing.

@baszalmstra , do you mean the entire concept or the name of the feature?

I feel having the ability to defer lockfile update after a series of operations on the manifest is useful. Pixi solve is fast, but not as fast as no solve at all :-)

**Comment by ruben-arts** [comment]
@olivier-lacroix I think what bas means is that we don't have the `--no-lock-update` flag as the functionality is the same as doing `--no-install --frozen`

**Comment by olivier-lacroix** [comment]
@baszalmstra @ruben-arts , I do not see these are equivalent. But maybe I misunderstand something?

see https://github.com/prefix-dev/pixi/issues/1669#issuecomment-2253657949

File: pixi.toml (hunk 2 above)
Comment (ruben-arts) [reply pixi.toml]: No, removed it

**Comment by ruben-arts** [comment]
@baszalmstra This PR does not yet remove the `--no-lockfile-update` but only fixes and refactors the code to behave like the documentation. Would you be okay with merging this before we refactor these flags based on the conclusion of this Discussion: https://github.com/prefix-dev/pixi/discussions/1707

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**